### ISSUE TITLE
[Feature]: Use kubernetes secrets for certificates

### ISSUE THREAD

**Issue by CanisLupusLupus** [Icinga/helm-charts#21]
### Affected Chart

icinga-stack

### Please describe your feature request

To complement issue #2 I propose that charts should use kubernetes secrets to provide certificates used for authentication in elasticsearch, gelf, etc.

E.g.: for elasticsearch certificates must be configured via `icinga2.features.elasticsearch.ca_path`, `icinga2.features.elasticsearch.cert_path` and `icinga2.features.elasticsearch.key_path` and certificates provided inside container (either directly in custom image or mounted via kustomized patch). I propose that certificates should be provided in secret and then mounted on default paths.

To illustrate change of config on elasticsearch: current config as part of `values.yaml`:
```
icinga2:
  features:
    elasticsearch:
      enable_tls: true # only used if TLS enabled
      ca_path: /etc/icinga2/pki/elastic-ca.crt
      cert_path: /etc/icinga2/pki/elastic-cert.crt
      key_path: /etc/icinga2/pki/elastic-cert.key
```
would become:
```
icinga2:
  features:
    elasticsearch:
      enable_tls: true # only used if TLS enabled
      caSecretKey: ca # key in existing secret
      certSecretKey: cert # key in existing secret
      keySecretKey: key # key in existing secret
```
which would then be projected in `icinga2` containers to default paths (`/etc/icinga2/pki/elastic-ca.crt`, `/etc/icinga2/pki/elastic-cert.crt` and `/etc/icinga2/pki/elastic-cert.key`).

I am also prepared to implement this feature and I am opening this issue to discuss it before the implementation.

**Comment by mocdaniel** [comment]
Sorry for getting back to you just now. I completely agree, being able to configure certificates via K8s secrets makes perfect sense, even more so after your previous PR.

If you want to work on this, feel free to open a PR with the suggested changes, I'll assign the issue to you for now.

**Comment by mocdaniel** [comment]
This has been implemented in #24
